### PR TITLE
Fix the way module candidate types are checked

### PR DESCRIPTION
**Pull request by feelingnothing**
# Summary
Fixes the way module candidate types are checked in `CommandsNext` extension

# Details
Changes the function [Type#IsAssignableFrom](https://docs.microsoft.com/en-us/dotnet/api/system.type.isassignablefrom) to a proper one [Type#IsSubclassOf](https://docs.microsoft.com/en-us/dotnet/api/system.type.issubclassof)
Current solution behaviour means that a class "could be a module" and this PR fixes this

### CODE REVIEW

**Comment by Emzi0767** [comment]
Derived check happens still, just later. This increases redundancy and potentially introduces new edge cases.

**Comment by feelingnothing** [comment]
> 
> 
> Derived check happens still, just later. This increases redundancy and potentially introduces new edge cases.

What type of edge cases we are talking about, in the documentation of the Type#IsAssingableFrom clearly says that class "can be assigned", existing code creates edge cases because if class cointains command and not a subclass of the module base class it should not be loaded because not all conditions are met with this type of class, there is an issue with current solution, and how the function that clearly shirks the range of types that could be loaded can create any edge cases

**Comment by Emzi0767** [comment]
Can I please get an English translation?

**Comment by feelingnothing** [comment]
> 
> 
> Can I please get an English translation?

Thank for pointing out my bad english skills, if you do not understand the certain part of the comment just quote me with it and i will try to clear it out for you

**Comment by Emzi0767** [comment]
I have no slightest clue what you were trying to say. Perhaps try to break it down into simpler, shorter sentences? As it stands it was a giant megasentence that is hard to parse.

**Comment by feelingnothing** [comment]
> 
> 
> I have no slightest clue what you were trying to say. Perhaps try to break it down into simpler, shorter sentences? As it stands it was a giant megasentence that is hard to parse.

In your original comment you said about edge cases that could potentially be met with my changes, but i could not understand the idea of this happening because the documentations of the Type#IsSubclassOf and Type#IsAssignableFrom are clearing out the potential unnessecery tries to load something thats not a module but contains a commands

**Comment by Emzi0767** [comment]
Ultimately though, there's either no difference at call site, or potential for other edge cases.

Assignability means it's directly descended or itself. The latter case is rejected later, where abstract and static classes are rejected, as well as anything that is not a class or a delegate. It doesn't account for conversions (which is good).

This means that there's potentially duplicated functionality, and potentially edge cases might arise from a change like that. There's more validation that would need to happen, for a chance with ultimately null effect

**Comment by feelingnothing** [comment]
Im apologizing about my attitude, clearly it works as you describe it, for my defense i can say that i did not get proper explanation in your first comment, but it is clear that i am not right in this situation, I would not say that my changes would not be correct, but now i understand the Type#IsAssignableFrom function more after your explanation, the name of the function confused me, that why i decided to bring this change, thank you for your time

**Comment by Emzi0767** [comment]
We're human after all
